data-prepper-plugins/geoip-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/configuration/EntryConfig.java

```diff
@@ -30,7 +44,48 @@ public String getTarget() {
         return target;
     }
 
-    public List<String> getFields() {
-        return fields;
+
+    public List<GeoIPField> getFields() {
+        if (!geoIPFields.isEmpty()) {
+            return geoIPFields;
+        }
+        if (!includeFields.isEmpty()) {
+            for(final String field: includeFields) {
+                final GeoIPField geoIPField = GeoIPField.findByName(field);
+                if (geoIPField != null) {
+                    geoIPFields.add(geoIPField);
+                }
+            }
+            return geoIPFields;
+        } else if (!excludeFields.isEmpty()) {
+            final List<GeoIPField> excludeGeoIPFields = new ArrayList<>();
+            for(final String field: excludeFields) {
+                final GeoIPField geoIPField = GeoIPField.findByName(field);
+                if (geoIPField != null) {
+                    excludeGeoIPFields.add(geoIPField);
+                }
+            }
+            final List<GeoIPField> values = new ArrayList<>(List.of(GeoIPField.values()));
+            values.removeAll(excludeGeoIPFields);
+            return values;
+        }
+        return geoIPFields;
+    }
+
+    public Set<GeoIPDatabase> getGeoIPDatabases() {
```

Comment:
I agree, I had it here because it simplifies getting this list for each entry, instead of calculation it in the processor.  But I can move it to processor initialization.

Reply:
I'd recommend a specific class for database determination. It will have a specific set of concerns and be more testable.